packages/graphback-cli/src/config/ProjectConfig.ts

```diff
@@ -0,0 +1,37 @@
+import { GraphQLGeneratorConfig } from 'graphback';
+import { FileConfig } from './FileConfig';
+
+/**
+ * Configuration for project that specifies every aspect of generation process.
+ * Sections:
+ * 
+ * - DB - database generation
+ * - graphqlCRUD - graphql CRUD generation
+ * - folders - input/outputs of the generator
+ */
+export interface ProjectConfig {
+    /**
+     * Database configuration
+     */
+    db: {
+        // tslint:disable-next-line: no-any
+        dbConfig: any,
+        database: string
+    },
+
+    /**
+     * GraphQL generator engine configuration
+     */
+    graphqlCRUD: GraphQLGeneratorConfig,
+
+    /**
+     * Configuration for input/output folders
+     */
+    folders: FileConfig
+
+    /**
+     * Empty object config that can be created to generate client
+     * Reserved for future usage
+     */
+    client: object
```

Comment:
```suggestion
    client: boolean
```

Reply:
One rule in building software that people depend on (libraries and frameworks) is to not introduce breaking changes at the beginning. I know that we assign boolean but if we will have any other field for the client later it will require people to change all their configs. 

So I have decided to use object as form of enablement and did not change underlying implementation from boolean.